packages/starlight/index.ts

```diff
@@ -7,25 +7,29 @@ import { starlightAsides } from './integrations/asides';
 import { starlightSitemap } from './integrations/sitemap';
 import { vitePluginStarlightUserConfig } from './integrations/virtual-user-config';
 import { errorMap } from './utils/error-map';
-import { StarlightConfigSchema, type StarlightUserConfig } from './utils/user-config';
+import { StarlightConfigSchema } from './utils/user-config';
 import { rehypeRtlCodeSupport } from './integrations/code-rtl-support';
+import { runPlugins, type StarlightUserConfigWithPlugins } from './utils/plugins';
 
-export default function StarlightIntegration(opts: StarlightUserConfig): AstroIntegration {
-	const parsedConfig = StarlightConfigSchema.safeParse(opts, { errorMap });
+export default function StarlightIntegration(
+	opts: StarlightUserConfigWithPlugins
+): AstroIntegration {
+	return {
+		name: '@astrojs/starlight',
+		hooks: {
+			'astro:config:setup': async ({ config, injectRoute, logger, updateConfig }) => {
+				const { integrations, userConfig } = await runPlugins(opts, logger);
+				const parsedConfig = StarlightConfigSchema.safeParse(userConfig, { errorMap });
```

Comment:
> Would be nice to add a comment regarding the order here.

Sure, I'll try to describe it better.

> I’m also wondering if we want to parse twice so we can error nicely before plugins run. I’m guessing currently we’d have the risk that a user provides bad config that causes some error in a plugin that Zod could catch but the user would see an error thrown by the plugin?

We definitely could extract the validation part in a function and run it before and after and it could indeed improve the experience when dealing with an error.